gamification/process/update.py

```diff
@@ -0,0 +1,142 @@
+
+    # Get the participant who opened this mr
+    mr_author = Participant.objects.get(username=mr.author)
+    mr_performed_at = mr.created_at
+    mr_updated_at = mr.updated_at
+
+    if mr.state == 'merged':
+        mr_labels = mr.labels.values('name')
+
+        # Get activity and points based on labels on the mr
+        mr_points, mr_activity_string = get_activity_with_points(
+            'merge_request', mr_labels)
+
+        # Update participant data
+        mr_author.add_points(mr_points,
+                             mr_activity_string,
+                             mr_performed_at,
+                             mr_updated_at,
+                             )
+        mr_author.save()
+
+        # Get all the issues this mr is closing
+        try:
+            issues = mr.get_closes_issues_object()
+        except Exception as ex:
+            logger.error(ex)
+            return
+        for issue in issues:
+            issue_labels = issue.labels.values('name')
+
+            # Get the participant who opened the issue
+            issue_author = Participant.objects.get(username=issue.author)
+            issue_performed_at = issue.created_at
+            issue_updated_at = issue.updated_at
+
+            # Get activity and points based on the labels on the issue
+            issue_points, issue_activity_string = get_activity_with_points(
+                'issue', issue_labels)
+
+            # Update participant data who opened the issue
+            issue_author.add_points(issue_points,
+                                    issue_activity_string,
+                                    issue_performed_at,
+                                    issue_updated_at,
+                                    )
+            issue_author.save()
+
+    elif mr.state == 'closed':
+        mr_points = -5
+        mr_activity_string = 'Merge request was closed without merge'
+        mr_author.deduct_points(mr_points,
+                                mr_activity_string,
+                                mr_performed_at,
+                                mr_updated_at,
+                                )
+        mr_author.save()
+
+
+def get_issue_objects():
+    """
+    Get issue objects saved in the database.
+
+    :return: a queryset of issue objects.
+    """
+    issues = Issue.objects.all()
+    return issues
+
+
+def update_participants_data_with_issue(issue):
+    """
+    Update participant data based on issue.
+
+    This method checks for 'status/invalid' or 'status/duplicate'
+    labels in the issue and based on that it based on that it
+    creates an activity and deduct 5 points.
+    """
+    labels = issue.labels.values('name')
+    labels_list = [label['name'] for label in labels]
+    issue_author = Participant.objects.get(username=issue.author)
+
+    if 'status/invalid' in labels_list or 'status/duplicate' in labels_list:
+        issue_points = -5
+        issue_activity = 'Created a invalid or duplicate issue'
```

Comment:
you can improve this activity description using the actual label which was encountered, and only when both appear, deciding which is more important in the label.

other activities use the convention

> 'Created a difficulty/newcomer type/bug issue'

here the same should happen.